google-cloud-core/src/main/java/com/google/cloud/GrpcTransportOptions.java

```diff
@@ -281,22 +261,34 @@ public int getMaxTimeout() {
     return maxTimeout;
   }
 
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
   @Override
-  protected int baseHashCode() {
-    return Objects.hash(super.baseHashCode(), executorFactoryClassName, initialTimeout,
+  public int hashCode() {
+    return Objects.hash(executorFactoryClassName, initialTimeout,
         timeoutMultiplier, maxTimeout);
   }
 
-  protected boolean baseEquals(GrpcServiceOptions<?, ?, ?> other) {
-    return super.baseEquals(other)
-        && Objects.equals(executorFactoryClassName, other.executorFactoryClassName)
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof GrpcTransportOptions)) {
```

Comment:
Ahh, you're right, I'll switch it. 

I think the implicit guideline is to use a `baseEquals` function for abstract classes, which assumes that more checks will be done in derived classes. According to http://stackoverflow.com/questions/596462/any-reason-to-prefer-getclass-over-instanceof-when-generating-equals , the leaf classes should do exact class comparison in `equals()`.